apps/partners-reference/lib/formatApplicationData.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  ApplicationUpdate,
+  ApplicantUpdate,
+  Language,
+  IncomePeriod,
+  ApplicationSubmissionType,
+  ApplicationStatus,
+  Address,
+} from "@bloom-housing/core"
+import moment from "moment"
+
+/*
+  Some of fields are optional, not active, so it occurs 'undefined' as value.
+  This function eliminates those fields and parse to a proper format.
+*/
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+
+function getAddress(condition, addressData) {
+  const blankAddress: Address = {
+    id: undefined,
+    createdAt: undefined,
+    updatedAt: undefined,
+    street: "",
+    street2: "",
+    city: "",
+    state: "",
+    zipCode: "",
+  }
+
+  return condition ? addressData : blankAddress
+}
+
+export const formatApplicationData = (data: any, listingId, editMode: boolean) => {
```

Comment:
Unfortunately, we can't, because the form can't return data in a proper BE format e.g. option can't return null, just "yes", "no" - for this reason, I created this helper and I think we should focus on proper types inside this "bridge" :)

I think we can type form, but I describe it in a proper comment.